six/projects/csm/source/SIXSensorModel.cpp

```diff
@@ -1110,5 +1111,21 @@ double SIXSensorModel::getCorrelationCoefficient(size_t cpGroupIndex,
                          "SIXSensorModel::getCorrelationCoefficient");
     }
 }
+
+DataType SIXSensorModel::getDataType(const csm::Des& des)
+{
+    std::string desid = des.subHeader().substr(NITF_DE_SZ, NITF_DESTAG_SZ);
+    str::trim(desid);
+    const std::string& subheader = des.subHeader();
+
+    const size_t desshsiOffset = 73 + 200;
+    const size_t desshsiLength = 60;
+    std::string desshsiField = des.subHeader().substr(
+            desshsiOffset, desshsiLength);
+    str::trim(desshsiField);
+
+    return NITFReadControl::getDataType(desid, subheader.size() - 200,
```

Comment:
Does this end up being the same as the first argument, then? I went ahead and removed it completely for now, if this is the case.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Friday, July 22, 2016 6:06 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Fix how CSM figures out an images type, to match how NITFReadControl now does it. (#90)


In six/projects/csm/source/SIXSensorModel.cpp<https://github.com/ngageoint/six-library/pull/90#discussion_r71952875>:

> @@ -1110,5 +1111,21 @@ double SIXSensorModel::getCorrelationCoefficient(size_t cpGroupIndex,

>                           "SIXSensorModel::getCorrelationCoefficient");

>      }

>  }

> +

> +DataType SIXSensorModel::getDataType(const csm::Des& des)

> +{

> +    std::string desid = des.subHeader().substr(NITF_DE_SZ, NITF_DESTAG_SZ);

> +    str::trim(desid);

> +    const std::string& subheader = des.subHeader();

> +

> +    const size_t desshsiOffset = 73 + 200;

> +    const size_t desshsiLength = 60;

> +    std::string desshsiField = des.subHeader().substr(

> +            desshsiOffset, desshsiLength);

> +    str::trim(desshsiField);

> +

> +    return NITFReadControl::getDataType(desid, subheader.size() - 200,

Well, since you already need to be dealing with this as a string anyhow... you can use the fourth argument and send in the DESID. I think this should be bytes [2, 27] of the string.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ngageoint/six-library/pull/90/files/b5fa7dce12abfbd03f84ce03aee18d4669cf64de#r71952875>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYItE0U3pUsg0tVsqBte6AoQSmzqQks5qYT7GgaJpZM4JNnoK>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.